psq/datastore_storage.py

```diff
@@ -40,16 +42,31 @@ def __init__(self, datastore):
         _check_for_thread_safety(datastore)
         self.datastore = datastore
 
+    @retry(
```

Comment:
I considered doing it that way but then I realized I was only doing it because the parameter names for the retrying library are overly verbose.  It's kind of a different way of repeating ourselves.  I can totally do a wrapper with the params baked in though when I get a minute.